FrenchQuizApp/Verbs/Views/VerbConjugationViewController.swift

```diff
@@ -0,0 +1,126 @@
+import UIKit
+
+class VerbConjugationViewController: UIViewController {
+    
+    var verb : Verbs?
+    var tense : String?
+    var conjugationDictionary : [String : [String : String]] = [ : ]
+    var tenseDictionary : [String : String] = [ : ]
+
+    var hasChanged = false
+    
+    @IBOutlet weak var frenchTextLabel: UILabel!
+    @IBOutlet weak var tenseTextLabel: UILabel!
+    @IBOutlet weak var tableView: UITableView!
+    
+    @IBAction func backTapped(_ sender: Any) {
+        dismiss(animated: true, completion: nil)
+    }
+    
+    override func viewDidLoad() {
+        guard let currentVerb = verb, let currentTense = tense else {
+            return
+        }
+        
+        frenchTextLabel.text = currentVerb.french
+        if let currentTense = tense {
+            tenseTextLabel.text = currentTense
+        }
+        conjugationDictionary = currentVerb.unarchiveDictionary()
+        for article in conjugationDictionary[currentTense]! {
+            tenseDictionary[article.key] = article.value
+            print(tenseDictionary)
+        }
+        
+        
+    }
+    
+    override func viewDidDisappear(_ animated: Bool) {
+        guard let currentTense = tense else {
+            return
+        }
+        
+        if verb != nil {
+            if hasChanged == true {
+                conjugationDictionary[currentTense] = tenseDictionary
+                let mutatedDictionary = verb!.mutateDictionaryToData(conjugationDictionary)
+                
+                verb!.conjugationDictionary = mutatedDictionary
+                ManagedData.saveContext()
+            }
```

Comment:
guard checks for nil, let unwraps verb. That should make it so you can remove the force-unwrap, `!`,  from this: verb!.mutateDictionaryToData()
```
guard let verb = verb else {
   return
}

if hasChanged {
   // do stuff
}
```